packages/m2m-gateway/src/services/eserviceService.ts

```diff
@@ -261,6 +303,63 @@ export function eserviceServiceBuilder(
     )({
       condition: isPolledVersionAtLeastResponseVersion(response),
     });
+  // eslint-disable-next-line max-params
+  async function assignEServiceDescriptorAttributesToGroup(
+    eserviceId: EServiceId,
+    descriptorId: DescriptorId,
+    groupIndex: number,
+    seed: m2mGatewayApi.EServiceDescriptorAttributesGroupSeed,
+    attributeKind: keyof catalogApi.Attributes,
+    headers: M2MGatewayAppContext["headers"]
+  ): Promise<void> {
+    const eservice = await retrieveEServiceById(headers, eserviceId);
+    const descriptor = retrieveEServiceDescriptorById(eservice, descriptorId);
+    const kindAttributeGroups = descriptor.attributes[attributeKind];
+    const attributeGroup = kindAttributeGroups.at(groupIndex);
+    if (!attributeGroup) {
+      throw eserviceDescriptorAttributeGroupNotFound(
+        attributeKind,
+        eserviceId,
+        descriptorId,
+        groupIndex
+      );
+    }
+    const updateAttributeGroup = [
+      ...attributeGroup,
+      ...seed.attributeIds.map((id) => ({
+        id,
+        explicitAttributeVerification: false,
+      })),
+    ];
+    const updatedGroups = [
+      ...kindAttributeGroups.slice(0, groupIndex),
+      updateAttributeGroup,
+      ...kindAttributeGroups.slice(groupIndex + 1),
+    ];
+    const configOptions = {
+      params: { eServiceId: eserviceId, descriptorId },
+      headers,
+    };
+    const response = await (descriptor.state ===
+    catalogApi.EServiceDescriptorState.Values.DRAFT
+      ? clients.catalogProcessClient.patchUpdateDraftDescriptor(
+          {
+            attributes: {
+              ...descriptor.attributes,
+              [attributeKind]: updatedGroups,
+            },
+          },
+          configOptions
+        )
+      : clients.catalogProcessClient.updateDescriptorAttributes(
+          {
+            ...descriptor.attributes,
+            [attributeKind]: updatedGroups,
+          },
```

Comment:
Extract this object as an `updatedAttributes` const and reuse it also in the other call and here